### PR TITLE
Pro 3517 view permission

### PR DESCRIPTION
**Pull request by haroun**
## Summary

Add support for `viewPermission`

## What are the specific steps to test this change?

1. `npm test`
2. Add `permission` or `viewPermission` to a widget schema and look for warnings in the server console
3. Add `permission` or `viewPermission` to a top level piece type schema field, you should see no warnings
4. Add `permission` or `viewPermission` to a sub level piece type schema field, you should see warnings in the server console
5. On the browser, open the `Network` panel and check the API response request when you open the manager modal for the piece having `permission` or `viewPermission` attributes. Try it with multiple credentials and verify that the fields is visible or not present for each case.

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: test/pieces.js [hunk 1]
```diff
@@ -1826,4 +1861,274 @@ describe('Pieces', function() {
...
+          'slug',
+          'visibility',
+          'archived',
+          'stock'
+        ];
+
+        assert.deepEqual(actual, expected);
+      });
+
+      it('should be able to retrieve fields with permission and viewPermission when having appropriate credentials', async function() {
```
Comment (boutell) [inline test/pieces.js:1998]: `permission` is not a test of view access, only `viewPermission` is.

**Review by boutell — CHANGES_REQUESTED** [review]
Almost there, just need to account for the fact that `field.permission` does not imply no view access via the API. That's why we have a separate `viewPermission` for cases where we want both behaviors.

You were right, though, to check the reverse: if you do fail `viewPermission` for the field it doesn't make sense to let you edit the field.

**Review by boutell — APPROVED** [review]

**Review by boutell — CHANGES_REQUESTED** [review]
This is actually approved, but I'm putting a hold on it to merge after this week's release just because it would be a little odd to add `viewPermission` before renaming `permission`.

**Comment by haroun** [comment]
@boutell the target branch for this PR is `feature/advanced-permission-2`, I'm merging it.
When we are ready to do a partial or full release, I'll create a new PR to merge `feature/advanced-permission-2` against `main`.

**Pull request merged**